### PR TITLE
arch/risc-v: Merge mcause.h into irq.h

### PR DESCRIPTION
**Pull request by no1wudi**
## Summary
Some exception relative definitions is already in irq.h, let's delete the mcause.h and refine these references.
## Impact
Rename only.
## Testing
CI and qemu-rv.

### CODE REVIEW

File: arch/risc-v/src/common/riscv_exception.c [hunk 1]
```diff
@@ -74,10 +72,9 @@ static const char *g_reasons_str[MCAUSE_MAX_EXCEPTION + 1] =
 
 void riscv_exception(uintptr_t mcause, uintptr_t *regs)
 {
-  uintptr_t cause = mcause & MCAUSE_INTERRUPT_MASK;
+  uint32_t cause = (uint32_t)mcause & RISCV_IRQ_MASK;
```
[the remark this answers is not in the document]
Comment (no1wudi) [reply arch/risc-v/src/common/riscv_exception.c:75]: mcause is always 32 bit on rv32 and rv64, so it's safe to do this cast and the _aleart below use the PRIx32 to format is.

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by pkarashchenko — APPROVED** [review]
LGTM!

**Comment by pkarashchenko** [comment]
@no1wudi seems that some additional include changes are needed to replace deleted header file

**Comment by no1wudi** [comment]
> @no1wudi seems that some additional include changes are needed to replace deleted header file

Done

**Pull request merged**